File: backend/src/main/java/de/micromata/lf8_3/model/Auto.java
package de.micromata.lf8_3.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Auto {

    @Id
    @GeneratedValue
    private long id;

    private String kennzeichen;

    private String hersteller;

    private String typ;

    private int baujahr;

    private int ps;

    private int ccm;

    private String farbe;

    private String kraftstoff;

    private int sitzplaetze;

    private String extras;

    private String zubehoer;

    private String versicherungsNr;

    private String tuev;

    private String asu;

    private Boolean verliehen;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "preisgruppe_fk")
    private Preisgruppe preisgruppe;

}

File: frontend/src/types/Auto.ts
import Preisgruppe from "./Preisgruppe";

export default interface Auto {
  id?: number;
  kennzeichen?: string;
  hersteller?: string;
  typ?: string;
  baujahr?: number;
  ps?: number;
  ccm?: number;
  farbe?: string;
  kraftstoff?: string;
  sitzplaetze?: number;
  extras?: string;
  zubehoer?: string;
  versicherungsNr?: string;
  tuev?: string;
  asu?: string;
  verliehen?: boolean;
  preisgruppe?: Preisgruppe;
}
